figures/montages/scripts/temporal_montages.py

```diff
@@ -0,0 +1,573 @@
+    except Exception as e:
+        print(f"Error processing well_fov {well_fov}: {e}")
+    output_df = pd.DataFrame(output_dict)
+    # sort by time
+    output_df = output_df.sort_values(by="time")
+    output_df.reset_index(drop=True, inplace=True)
+    return output_df
+
+
+# ## Pathing and preprocessing
+
+# In[3]:
+
+
+umap_file_path = pathlib.Path(
+    "../../../3.generate_umap_and_PCA/results/UMAP/single-cell_profiles_CP_scDINO_umap.parquet"
+).resolve(strict=True)
+large_montage_white_background_output_dir = pathlib.Path(
+    "../figures/large_montage_white_background.png"
+).resolve()
+large_montage_black_background_output_dir = pathlib.Path(
+    "../figures/large_montage_black_background.png"
+).resolve()
+mini_montage_white_background_output_dir = pathlib.Path(
+    "../figures/mini_montage_white_background.png"
+).resolve()
+mini_montage_black_background_output_dir = pathlib.Path(
+    "../figures/mini_montage_black_background.png"
+).resolve()
+mini_montage_black_background_output_dir.parent.mkdir(parents=True, exist_ok=True)
+umap_df = pd.read_parquet(umap_file_path)
+# make the time column numeric
+umap_df["Metadata_Time"] = pd.to_numeric(umap_df["Metadata_Time"])
+umap_df["Metadata_Time"] = umap_df["Metadata_Time"].astype(int)
+umap_df["Metadata_Time"] = umap_df["Metadata_Time"] * 30
+umap_df["Metadata_Well_FOV"] = (
+    umap_df["Metadata_Well"].astype(str) + "_" + umap_df["Metadata_FOV"].astype(str)
+)
+
+
+# ## Get dfs containing cell tracks
+
+# In[4]:
+
+
+# get a random Metadata_track_id for a few wells
+well_fovs = umap_df["Metadata_Well_FOV"].unique()
+# for each well we will find a Metadata_track_id that has
+# all time points
+track_ids = {"well_fovs": [], "track_id": []}
+np.random.seed(1)
+for well_fov in well_fovs:
+    well_df = umap_df[umap_df["Metadata_Well_FOV"] == well_fov]
+    track_id_counts = well_df["Metadata_track_id"].value_counts()
+    # find a track id that has all time points (48)
+    full_track_ids = track_id_counts[track_id_counts == 13].index.tolist()
+
+    for track_id in full_track_ids:
+        track_ids["well_fovs"].append(well_fov)
+        track_ids["track_id"].append(track_id)
+
+
+# In[5]:
+
+
+df = pd.DataFrame(track_ids)
+df = df.sort_values(by="well_fovs")
+df.head()
+
+
+# In[6]:
+
+
+c02_df = generate_image_pannel_df(umap_df, "C-02_0001", 23)  # DMSO
+c03_df = generate_image_pannel_df(umap_df, "C-03_0001", 28)  # 0.61
+c04_df = generate_image_pannel_df(umap_df, "C-04_0001", 35)  # 1.22
+c05_df = generate_image_pannel_df(umap_df, "C-05_0001", 44)  # 2.44
+c06_df = generate_image_pannel_df(umap_df, "C-06_0001", 106)  # 4.88
+c07_df = generate_image_pannel_df(umap_df, "C-07_0001", 109)  # 9.77
+c08_df = generate_image_pannel_df(umap_df, "C-08_0001", 35)  # 19.23
+c09_df = generate_image_pannel_df(umap_df, "C-09_0001", 22)  # 39.06
+c10_df = generate_image_pannel_df(umap_df, "C-10_0001", 101)  # 78.13
+c11_df = generate_image_pannel_df(umap_df, "C-11_0001", 141)  # 156.25
+
+
```

Comment:
I am a bit confused here, if you are getting random tracks in df, why use UMAP_df then to generate the image panels? I would consider changing df to a more explicit name and explaining how you got those cell IDs you are specifiying in the function.